### DISCUSSION TITLE
Is there a "Portable mode"?

### DISCUSSION THREAD

**Discussion by TheBestPessimist** [seerge/g-helper#628] [Q&A] [answered]
Basically, i'd like it if GHelper would store all its data (logs and config json. anything else?) in the installation folder.

I keep a bunch of portable apps in the same folder (any app which can be portable, i use it in that way), which i backup regularly, and this would make my backups easier.

**Comment by seerge** [comment]
@TheBestPessimist i may consider adding that, but then you will have to "define" if app should use exe folder (or appdata folder as now) for storing config. It doesn't store anything else besides that honestly.

**Reply by TheBestPessimist** [reply]
One way you could make it work is: if where the exe is you also find a `config.json` file, then assume it's portable mode and use that one. If there is no config file, then use the one in app data.
With that logic you can start, just like now, with app data storage, as it's assumed by everybody else, and those who want portable, can move the config file to app folder.

**Reply by seerge** [reply, marked as answer]
Try this build 
[GHelper.zip](https://github.com/seerge/g-helper/files/11756167/GHelper.zip)

But keep in mind - you will have to either copy (or at least create empty) ``config.json`` in same folder as app to make it work

Logic : https://github.com/seerge/g-helper/blob/28e949bee4e428987d1e8ef2cd0d76bea071b31f/app/AppConfig.cs#L21

**Reply by TheBestPessimist** [reply]
1. Logs are still in appdata ❌ 
2. appdata/ghelper folder is still created ❌ 
3. it uses the config from portable folder ✅

**Reply by seerge** [reply]
@TheBestPessimist 

1. Logs are not part of the user-data :) it's mainly for debug. So i don't see point in changing that as well.
2. Folder is created cause it's needed for logs :)
